docs/examples/rx/05_read_storage/index.js

```diff
@@ -46,4 +48,47 @@ function main () {
     });
 }
 
+function mainTap () {
+    new ApiRx(provider).isReady
+      .pipe(
+        // Here we ake our basic chain state/storage queries, all in one go
+        switchMap((api) => combineLatest(
+          // We're combining the results together with the emitted value 'api',
+          // which we're turning back into an observable using of()
+          of(api),
+          api.query.session.validators().pipe(first()),
+          api.query.system.accountNonce(ALICE).pipe(first()),
+          api.query.timestamp.blockPeriod().pipe(first())
+        )),
+        tap(([,, accountNonce, blockPeriod]) => {
+          // We're using tap() to print out the results of our account nonce and block period queries
+          console.log(`accountNonce(${Alice}) ${accountNonce}`);
+          console.log(`blockPeriod ${blockPeriod.toNumber()} seconds`);
+
+        }),
+        switchMap(([api, validators]) => {
+          // In the next step, we're checking if the node has active validators.
+          // If it does, we're making another call to the api to get the balances for all validators
+          const balances = (validators && validators.length > 0)
+            ? combineLatest(validators.map(authorityId => api.query.balances.freeBalance(authorityId).pipe(first())))
+            : of(null);
+          // If there are no validators, we're not returning any balances
+          return combineLatest(
+            of(validators),
+            balances
+          );
+        })
+      )
+      // Then we're subscribing to the emitted results
+      .subscribe(([validators, validatorBalances]) => {
+        if (validatorBalances) {
+          // And lastly we print out the authorityIds and balances of all validators
+          console.log('validators', validators.map((authorityId, index) => ({
+            address: authorityId.toString(),
+            balance: validatorBalances[index].toString()
+          })));
+        }
+      });
+}
+
```

Comment:
Not sure why we don't just exit after subscribe - we use first here everywhere (different from the promise version I believe), so are only returning single results. So We can just exit after all results.